src/components/CsvUpload/CsvUpload.tsx

```diff
@@ -0,0 +1,138 @@
+      <Spacer />
+      <IconButton icon={<CloseIcon size={16} />} onClick={onClearFile} />
+    </HStack>
+  )
+}
+
+type CsvUploadProps = {
+  file: File | null
+  onFileSelected: (file: File | null) => void
+}
+export const CsvUpload = ({ file, onFileSelected }: CsvUploadProps) => {
+  const [errorMessage, setErrorMessage] = useState<string | undefined>()
+  const fileInputRef = useRef<HTMLInputElement | null>(null)
+
+  const handleBrowseClick = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault()
+    fileInputRef.current?.click()
+  }
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files
+    if (!files || files.length === 0) return
+
+    const fileArray = Array.from(files)
+    onDrop(fileArray)
+  }
+
+  const onDrop = useCallback(
+    ([firstFile, ...restFiles]: File[], rejections: FileRejection[] = []) => {
+      const hasTooManyFiles = rejections.some(r => r.errors.some(e => e.code === 'too-many-files'))
+      if (restFiles.length > 0 || hasTooManyFiles) {
+        onFileSelected(null)
+        setErrorMessage('Too many files selected')
+        return
+      }
+
+      if (rejections.length > 0) {
+        onFileSelected(null)
+        setErrorMessage('Unknown upload error')
+        return
+      }
+
+      const maybeErrorMessage = validateCsvFile(firstFile)
+      if (!maybeErrorMessage) {
+        onFileSelected(firstFile)
+        setErrorMessage(undefined)
+      }
+      else {
+        onFileSelected(null)
+        setErrorMessage(maybeErrorMessage)
+      }
+    },
+    [onFileSelected],
+  )
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop,
+    multiple: false,
+    noClick: true,
+    noKeyboard: true,
+  })
+
+  return (
+    <VStack gap='xs'>
+      <VStack
+        className={classNames(
+          'Layer__csv-upload',
+          { 'Layer__csv-upload--drag-active': isDragActive },
+        )}
+        align='center'
+        justify='center'
+        gap='xs'
+        {...getRootProps()}
+      >
+        <input {...getInputProps()} />
+        <HStack align='center' gap='xs'>
+          <UploadCloud size={12} />
+          <P size='sm'>
+            {'Drag and drop a file, or '}
+            <button className='Layer__csv-upload__browse-link' onClick={handleBrowseClick}>
+              Browse
+            </button>
+          </P>
+          <input type='file' ref={fileInputRef} style={{ display: 'none' }} accept='.csv' onChange={handleFileChange} />
+        </HStack>
+        <P size='sm' variant='subtle'>Upload file in CSV format</P>
+        {errorMessage && (
+          <DataState
+            className='Layer__csv-upload__error-message'
+            status={DataStateStatus.failed}
+            title='Cannot upload file'
+            description={errorMessage}
+            inline
+          />
+        )}
+      </VStack>
+      {file && <FileRow file={file} onClearFile={() => onFileSelected(null)} />}
```

Comment:
![image](https://github.com/user-attachments/assets/6173eeab-1f63-4ef7-9015-20dafd2bc14b)

The first time I got into this state where I have an upload (below) but still have the CTA to to upload a file I got confused for a second and thought I hadn't uploaded anything successfully. What do you think about hiding the upload drag & drop box and changing it to just be the uploaded file? (which you can obviously delete and start over)